src/nupic/algorithms/ColumnPooler.cpp

```diff
@@ -227,6 +232,16 @@ class ColumnPooler // : public Serializable
   }
 
 
+  Real initProximalPermanence(Real connectedPct = 0.5f) {
+    // TODO: connectedPct as a true parameter? Maybe? Someday?
```

Comment:
why not do it properly? Could be passed from constructor

Reply:
Even better, this whole function could be an argument.  I'm thinking something like:
```
typedef function<Permanence(Random&)> InitialPermanence_t;

InitialPermanence_t DefaultInitProximalPermanence( args ) {
  return [=](Random &rng) -> Permanence {
    return rng.getReal64(); // placeholder
  };
}
```
The default parameters for the column pooler would have an empty function (std::function has a standard way to check if it contains a valid function or is empty).  Detect no argument given, and create this default with the correct connectedThreshold.

Then the user can supply their own function!